File: src/main/java/org/example/Repository/InscriereDBRepository.java
package org.example.Repository;

import org.example.Domain.Inscriere;
import org.example.Domain.Pair;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

import java.util.List;

public class InscriereDBRepository implements InscriereRepositoryInterface {
    private static final Logger logger = LogManager.getLogger(CursaDBRepository.class);
    private JdbcUtils dbUtils;

    public InscriereDBRepository(Properties properties) {
        logger.info("Initializing AngajatDBRepository with properties: {}", properties);
        dbUtils = new JdbcUtils(properties);
    }
    @Override
    public Pair<Long, Long> insert(Inscriere entity) {

        return null;
    }

    @Override
    public void updateById(Pair<Long, Long> longLongPair, Inscriere entity) {

    }

    @Override
    public void deleteById(Pair<Long, Long> longLongPair) {

    }

    @Override
    public Inscriere getById(Pair<Long, Long> longLongPair) {
        return null;
    }

    @Override
    public List<Inscriere> getAll() {
        return List.of();
    }
}

File: Repository/InscriereDBRepository.cs
using log4net;
using Motociclete.Domain;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Data.Sqlite;

namespace Motociclete.Repository
{
    public class InscriereDBRepository : IRepositoryInscriere
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(InscriereDBRepository));
        private readonly DBUtils _dbUtils;
        private readonly ParticipantDBRepository _pRepo;
        private readonly EchipaDBRepository _eRepo;

        public InscriereDBRepository(IDictionary<string, string> properties, ParticipantDBRepository pRepo, EchipaDBRepository eRepo)
        {
            _dbUtils = new DBUtils(properties);
            _pRepo = pRepo;
            _eRepo = eRepo;
            logger.Info("InscriereDBRepository initialized");
        }

        public Pair<long, long> Insert(Inscriere entity)
        {
            logger.Info("Inserting new Inscriere");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "INSERT INTO Inscriere (cursa_id, participant_id) VALUES (@cursa_id, @participant_id)";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@cursa_id", entity.Cursa.id);
                            cmd.Parameters.AddWithValue("@participant_id", entity.Participant.id);

                            cmd.ExecuteNonQuery();
                            logger.Info("Inserted new Inscriere successfully");
                            return entity.id;
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error inserting Inscriere: " + ex.Message);
                }
            }
            return null;
        }

        public void UpdateById(Pair<long, long> id, Inscriere entity)
        {
            // Not implemented
        }

        public void DeleteById(Pair<long, long> id)
        {
            // Not implemented
        }

        public Inscriere GetById(Pair<long, long> id)
        {
            // Not implemented
            return null;
        }

        public List<Inscriere> GetAll()
        {
            // Not implemented
            return new List<Inscriere>();
        }

        public List<Participant> GetParticipantiInscrisiByCursaId(long id)
        {
            logger.Info("Fetching Participants for CursaId: " + id);
            List<Participant> participants = new List<Participant>();

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "SELECT * FROM Inscriere WHERE cursa_id = @cursa_id";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@cursa_id", id);
                            using (var reader = cmd.ExecuteReader())
                            {
                                while (reader.Read())
                                {
                                    participants.Add(_pRepo.GetById(reader.GetInt64(reader.GetOrdinal("participant_id"))));
                                }
                            }
                        }
                        logger.Info("Participants fetched successfully");
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error fetching Participants: " + ex.Message);
                }
            }
            return participants;
        }
    }

}
